core/src/consensus/stake/mod.rs

```diff
@@ -126,6 +127,20 @@ impl ActionHandler for Stake {
         action.verify(current_params)
     }
 
+    fn key_allowance(&self, bytes: &[u8]) -> KeyAllowance {
```

Comment:
This makes you decode one more time for every custom action transactions.
What I thought is rejecting the transaction in [`self_nominate`](https://github.com/CodeChain-io/codechain/blob/39484fb70c7f15c625ee572676209e9fe613e2fd/core/src/consensus/stake/mod.rs#L202) if the sender address doesn't match the sender public key.

Reply:
FYI, `sender` of `self_nominate` cannot be the address of the `sender_public`.
`sender` is always the address of the master key address and `sender_public` is the public key that signed the transaction.
So if the transaction is signed with the regular key, that can be different.
I recommend you to rename `sender` to `fee_payer` and compare them to check whether the transaction is signed with the master key or not.